### PR TITLE
Add common Transport code

### PR DESCRIPTION
**Pull request by pcppcp**

### CODE REVIEW

File: raiden_libs/transport/matrix.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        while self.is_running.is_set() is False:
+            try:
+                self.connect()
+                self.room.add_listener(lambda room, event: self.dispatch(room, event))
+                self.sync_history()
+                self.do_reconnect.wait()
+                if self.do_reconnect.get() == 100:
+                    gevent.sleep(self.retry_timeout)
+                    continue
+            except (requests.exceptions.ConnectionError,
```
Comment (palango) [inline raiden_libs/transport/matrix.py]: Does that fit on a line? Otherwise linebreak after `(`

File: raiden_libs/transport/matrix.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    MatrixHttpLibError
+                    ) as e:
+                log.warn("Connection to %s failed. Retrying in %d seconds (%s)" %
```
Comment (palango) [inline raiden_libs/transport/matrix.py]: Formatting as well.

File: requirements.txt [hunk 3]
```diff
@@ -1,7 +1,8 @@
 eth-utils>=1.0.1,<2.0
 coincurve>=7.1.0,<8.0
 web3>=4.0.0-beta.13,<5.0
```
Comment (palango) [inline requirements.txt]: Can you update to 4.0.0 while you're at it?

**Comment by pcppcp** [comment]
`py-evm` version in `requirements.txt` can be unfreezed once https://github.com/ethereum/eth-tester/pull/77 is merged

**Pull request merged**